csaf/src/entry_point/mod.rs

```diff
@@ -0,0 +1,93 @@
+use crate::source::HttpSourceError;
+use sectxtlib::SecurityTxt;
+use url::{ParseError, Url};
+use walker_common::fetcher;
+use walker_common::fetcher::Fetcher;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Fetch error: {0}")]
+    Fetcher(#[from] fetcher::Error),
+    #[error("URL error: {0}")]
+    Url(#[from] ParseError),
+    #[error("securityTxtParseError error: {0}")]
+    SecurityTextError(#[from] sectxtlib::ParseError),
+}
+
+impl From<Error> for HttpSourceError {
+    fn from(value: Error) -> Self {
+        match value {
+            Error::Fetcher(err) => Self::Fetcher(err),
+            Error::Url(err) => Self::Url(err),
+            Error::SecurityTextError(err) => Self::SecurityTextError(err),
+        }
+    }
+}
+pub struct EntryPoint {
+    pub provider_metadata_url: Url,
+}
+
+impl EntryPoint {
+    pub async fn retrieve_entry_point_from_security_text(
+        fetcher: &Fetcher,
+        base_url: Url,
+    ) -> Result<Self, Error>
+    where
+        Self: Sized + Send,
+    {
+        let mut provider_metadata_url = "".to_string();
+        if let Some(host_url) = base_url.host_str() {
+            let host_url_string = format!("https://{}", host_url);
+            let host_url = Url::parse(host_url_string.as_str())?;
+            log::warn!(
+                "Querying provider metadata url from security text of {:?}",
+                host_url_string
+            );
+            Self::get_url(
+                fetcher,
+                &mut provider_metadata_url,
+                host_url.join("/.well-known/security.txt")?,
+            )
+            .await?;
+
+            let provider_metadata_url = Url::parse(provider_metadata_url.as_str())?;
+            Ok(Self {
+                provider_metadata_url,
+            })
+        } else {
+            Err(Error::Url(ParseError::EmptyHost))
+        }
+    }
+
+    async fn get_url(
+        fetcher: &Fetcher,
+        provider_metadata_url: &mut String,
```

Comment:
Returning a value through a `&mut` feels a bit odd. Is there a reason for doing so? Compared to using the return?

I am just wondering what the state of the variable would be in case of an error? Or when there are multiple values found?